core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java

```diff
@@ -240,7 +240,7 @@ public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asJson(Class<
                                                                               ObjectMapper mapper) {
         requireNonNull(clazz, "clazz");
         requireNonNull(mapper, "mapper");
-        return asEntity(ResponseAs.json(clazz));
+        return asEntity(ResponseAs.json(clazz, mapper));
```

Comment:
Could you also add a unit test to detect unexpected changes in the future?

Reply:
I have added the test code in WebClientRequestPreparationTest.
You can review the changes in the following commit: 8bdcb3e.

However, I am not sure it helps detect unexpected changes in the future.
Could you let me know which test cases should be added?